tests/benchmarks/latency_measure/README.rst

```diff
@@ -29,81 +33,125 @@ threads:
-        Take a semaphore (no blocking) from user thread     :     315 cycles ,     2625 ns :
-        Take a semaphore (context switch K -> K)            :    1340 cycles ,    11167 ns :
-        Give a semaphore (context switch K -> K)            :    1460 cycles ,    12167 ns :
-        Take a semaphore (context switch K -> U)            :    1540 cycles ,    12838 ns :
-        Give a semaphore (context switch U -> K)            :    1800 cycles ,    15000 ns :
-        Take a semaphore (context switch U -> K)            :    1690 cycles ,    14084 ns :
-        Give a semaphore (context switch K -> U)            :    1650 cycles ,    13750 ns :
-        Take a semaphore (context switch U -> U)            :    1890 cycles ,    15756 ns :
-        Give a semaphore (context switch U -> U)            :    1990 cycles ,    16583 ns :
-        Lock a mutex from kernel thread                     :     105 cycles ,      875 ns :
-        Unlock a mutex from kernel thread                   :      17 cycles ,      141 ns :
-        Lock a mutex from user thread                       :     330 cycles ,     2750 ns :
-        Unlock a mutex from user thread                     :     255 cycles ,     2125 ns :
-        Average time for heap malloc                        :     606 cycles ,     5058 ns :
-        Average time for heap free                          :     422 cycles ,     3516 ns :
+        *** Booting Zephyr OS build zephyr-v3.5.0-3537-g5dbe0ce2622d ***
+        THREAD yield.preemptive.ctx.(K -> K)                :     990 cycles ,     8250 ns :
+        THREAD yield.preemptive.ctx.(U -> U)                :    1285 cycles ,    10712 ns :
+        THREAD yield.preemptive.ctx.(K -> U)                :    1178 cycles ,     9817 ns :
+        THREAD yield.preemptive.ctx.(U -> K)                :    1097 cycles ,     9145 ns :
+        THREAD yield.cooperative.ctx.(K -> K)               :     990 cycles ,     8250 ns :
+        THREAD yield.cooperative.ctx.(U -> U)               :    1285 cycles ,    10712 ns :
+        THREAD yield.cooperative.ctx.(K -> U)               :    1178 cycles ,     9817 ns :
+        THREAD yield.cooperative.ctx.(U -> K)               :    1097 cycles ,     9146 ns :
+        ISR resume.interrupted.thread.kernel                :    1120 cycles ,     9333 ns :
+        ISR resume.different.thread.kernel                  :    1010 cycles ,     8417 ns :
+        ISR resume.different.thread.user                    :    1207 cycles ,    10062 ns :
+        THREAD create.kernel.from.kernel                    :     955 cycles ,     7958 ns :
+        THREAD start.kernel.from.kernel                     :    1095 cycles ,     9126 ns :
+        THREAD suspend kernel.from.kernel                   :    1064 cycles ,     8874 ns :
+        THREAD resume kernel.from.kernel                    :     999 cycles ,     8333 ns :
+        THREAD abort kernel.from.kernel                     :    2280 cycles ,    19000 ns :
```

Comment:
> Changes the summary lines used in the latency_measure benchmark
so that they follow a consistent pattern.

>    OBJECT  action.<brief details about action>

> This makes it easier to locate results belonging to a particular object group. It also has the benefit to keep results colocated should the output be sorted by an external tool.


Some records have `OBJECT action.does.something` (e.g. `ISR resume.interrupted.thread.kernel`)
others - `OBJECT does.something.action` (e.g. `THREAD create.kernel.from.kernel`)

In this way these record might be unexpectedly mixed, if sorted in ascii, e.g.
```
THREAD create.kernel.from.kernel
THREAD create.user.from.user
...
THREAD start.kernel.from.kernel
THREAD start.user.from.user
```

Reply:
Since this is opening the door to multiple orderings, there is arguably no ideal way to order it all. One one hand, the raw output of the benchmark groups the items according to what is easiest and most logical for the test. Threads, for example, have a natural order to them--they are created, started, get acted upon (suspend/resume) and eventually terminate/abort.

Once sorting by an external tool is introduced, at this level of events, we can only make a guess about how the tool will sort things--alphabetical does seem like a reasonable prediction. With that sorting in mind, all the "THREAD start.\*" would be co-located, just as all the "SEMAPHORE give.\*" would be co-located and it would be easier to see how the different details/conditions affect the action.

All that being said, I would prefer to continue using the _OBJECT action.'brief details about action'_ output model.

Reply:
> All that being said, I would prefer to continue using the _OBJECT action.'brief details about action'_ output model.

agree, when there is consistency in 'OBJECT action.some.details.there it is up to an external tool (like ELK) to parse and sort as it needs.